### PR TITLE
Rename dist target to cdn

### PR DESCRIPTION
**Pull request by unDemian**
**Built on top of https://github.com/Automattic/happychat-client/pull/177**

## Summary
Target `dist` was renamed to `cdn` because it does expose the happychat api to the window object (dist can be too generic especially as we add npm as a target also)

## Testing
Code review only.

### CODE REVIEW

File: webpack.browser.config.js [hunk 1]
```diff
@@ -5,9 +5,9 @@ const UglifyJsPlugin = require( 'uglifyjs-webpack-plugin' );
 const LodashModuleReplacementPlugin = require( 'lodash-webpack-plugin' );
 
 module.exports = {
-	entry: './dist/index.js',
+	entry: './targets/browser/index.js',
 	output: {
-		filename: './targets/dist/happychat.js',
+		filename: './targets/browser/happychat.js',
```
Comment (lamosty) [inline webpack.browser.config.js]: 👍

File: webpack.browser.config.js [hunk 2]
```diff
@@ -5,9 +5,9 @@ const UglifyJsPlugin = require( 'uglifyjs-webpack-plugin' );
 const LodashModuleReplacementPlugin = require( 'lodash-webpack-plugin' );
 
 module.exports = {
-	entry: './dist/index.js',
```
Comment (lamosty) [inline webpack.browser.config.js]: 👍

**Comment by chriskmnds** [comment]
LGTM

**Comment by nosolosw** [comment]
For some more context, I wrote https://github.com/Automattic/happychat-client/pull/177#issuecomment-377801589 before reviewing this PR. Now that I also see this, how about: 

- keeping the `dist` target, but to only contain the production CSS for the reasons I've mentioned [before](https://github.com/Automattic/happychat-client/pull/177#issuecomment-377801589). `dist` is quite standard for this kind of things, but I wouldn't mind if we choose perhaps `cdn` if that makes its purpose more obvious.
- reorganizing the `standalone` target to have:
  - `targets/dist/standalone/index.js`: what you had in `browser`.
  - `targets/dist/standalone/example/`: move here the current example with the oauth wpcom mechanism.

Just a nitpick: `browser` feel too generic for me, all existing targets are for browsers! Repurposing `standalone` as I mentioned above was long overdue on my part and created some confusion. We can fix this now :)

**Comment by unDemian** [comment]
> keeping the dist target, but to only contain the production CSS for the reasons I've mentioned before. dist is quite standard for this kind of things, but I wouldn't mind if we choose perhaps cdn if that makes its purpose more obvious.

I agree that `dist` is used for production ready files, but as we move forward and have this served as an npm package also, it might become a bit confusing because `dist` is used in npm package as well.

> reorganizing the standalone target to have:

I'm not sure this is needed, is standalone any different from `cdn` then ? If we move the authentication in an example the target is not really "standalone" as it needs the auth in the example to work.


> Just a nitpick: browser feel too generic for me, all existing targets are for browsers! Repurposing standalone as I mentioned above was long overdue on my part and created some confusion. We can fix this now :)

That is correct right now, but we are adding npm as a target also. But maybe `cdn` would be a better fit.

**Comment by nosolosw** [comment]
The `dist/` / `standalone/` confusion lays on me. I don't think the WordPress OAuth authentication should be part of the `standalone` target, I only did that for the example to work because authentication is owned by the hosts, not the library. But! If we're updating targets I think it'll be a good idea to fix that confusion and use `standalone` for what you call `browser`.

After seeing all your work and the idea of separating dev/production scripts, I don't think the `dist` target makes sense anymore.

**Comment by unDemian** [comment]
That makes sense, I'll update merge these PRs and create a new one that will clarify and clean the targets.

**Pull request merged**